src/hooks/useSignErrorHandler.ts

```diff
@@ -0,0 +1,62 @@
+import { t } from '@lingui/macro';
+import { useEffect } from 'react';
+
+import useToastMsg from './useToastMsg';
+import type { VCSignError } from './useVerifiableCredential';
+import { VCSignErrorType } from './useVerifiableCredential';
+
+export type CustomErrorProps = {
+  [key: string]: {
+    title: string;
+    description: string;
+  };
+};
+
+/**
+ * A Hook to handle sign error and display the error message in toast.
+ *
+ * @param signError - The sign error.
+ * @param errorMessages - The override error messages.
+ */
+export function useSignErrorHandler(
+  signError: VCSignError | undefined,
+  errorMessages?: CustomErrorProps,
+) {
+  const { showErrorMsg } = useToastMsg();
+
+  useEffect(() => {
+    if (signError) {
+      const defaultTitle = t`Unknown Error`;
+      const defaultDescription = t`Unknown Error`;
+
+      const errorMsg = errorMessages ?? {};
```

Comment:
need to create a new errorMsg variable to prevent the type lint error on errorMessages.